app/src/main/java/com/nutomic/syncthingandroid/service/EventProcessor.java

```diff
@@ -94,6 +94,12 @@ public void onDone(long lastId) {
     @Override
     public void onEvent(Event event) {
         switch (event.type) {
+            case "ConfigSaved":
+                if (mApi != null) {
+                    Log.v(TAG, "Forwarding ConfigSaved event to RestApi to get the updated config.");
+                    mApi.onWebGuiAvailable();
```

Comment:
We should probably just have a mApi.reloadConfig and call that. It seems that on web ui available calls a bunch of other endpoints that have not changed and does some magic atomic increments.

```
        new GetRequest(mContext, mUrl, GetRequest.URI_CONFIG, mApiKey, null, result -> {
            Log.v(TAG, "reloadConfig: " + result);
            mConfig = new Gson().fromJson(result, Config.class);
            if (mConfig == null) {
                throw new RuntimeException("config is null: " + result);
            }
        });
```

Reply:
@Nutomic Could you please give me some advice if I need the full code triggering all classes to listen to the new, reloaded config as if the syncthing app restart option had been fired? Reading through the classes, I thought it would be the best way to fully perform a reinit of all associated classes (like when onWebGuiAvailable() regularly fires) to get the right increment counter for reading events, folderstatus model and so on. If we change the webgui port and address, it looks wrong to me just pushing a new config in and missing some steps. I hope you can help me clarify that.
@AudriusButkevicius Strategically I like your approach of minimalistic reloading the config and associated classes. Let's wait what Nutomic says and then decide how to generate the mApi.reloadConfig() function.

Reply:
I can tell you from reading the code that you don't need that, as it fetches system info and device id, which is not changed by saving the config.

Reply:
Port and what not is not handled by the existing code anyway

Reply:
Ok, agreed. I'll make a new suggestion.